tests/contrib/config/test_templated_config.py

```diff
@@ -0,0 +1,340 @@
+    proj_catalog = tmp_path / "base" / "catalog.yml"
+    _write_yaml(proj_catalog, param_config_exceptional)
+
+
+class TestTemplatedConfigLoader:
+    @pytest.mark.usefixtures("proj_catalog_param")
+    def test_catlog_parameterized_w_dict(self, tmp_path, conf_paths, template_config):
+        """Test parameterized config with input from dictionary with values"""
+        (tmp_path / "local").mkdir(exist_ok=True)
+
+        catalog = TemplatedConfigLoader(conf_paths, globals_dict=template_config).get(
+            "catalog*.yml"
+        )
+
+        assert catalog["boats"]["type"] == "SparkDataSet"
+        assert (
+            catalog["boats"]["filepath"] == "s3a://boat-and-car-bucket/01_raw/boats.csv"
+        )
+        assert catalog["boats"]["columns"]["id"] == "VARCHAR"
+        assert catalog["boats"]["columns"]["name"] == "VARCHAR"
+        assert catalog["boats"]["columns"]["top_speed"] == "FLOAT"
+        assert catalog["boats"]["users"] == ["fred", "ron"]
+
+    @pytest.mark.usefixtures("proj_catalog_param", "proj_catalog_globals")
+    def test_catlog_parameterized_w_globals(self, tmp_path, conf_paths):
+        """Test parameterized config with globals yaml file"""
+        (tmp_path / "local").mkdir(exist_ok=True)
+
+        catalog = TemplatedConfigLoader(conf_paths, globals_pattern="*globals.yml").get(
+            "catalog*.yml"
+        )
+
+        assert catalog["boats"]["type"] == "SparkDataSet"
+        assert (
+            catalog["boats"]["filepath"] == "s3a://boat-and-car-bucket/01_raw/boats.csv"
+        )
+        assert catalog["boats"]["columns"]["id"] == "VARCHAR"
+        assert catalog["boats"]["columns"]["name"] == "VARCHAR"
+        assert catalog["boats"]["columns"]["top_speed"] == "FLOAT"
+        assert catalog["boats"]["users"] == ["fred", "ron"]
+
+    @pytest.mark.usefixtures("proj_catalog_param")
+    def test_catlog_parameterized_no_params(self, tmp_path, conf_paths):
+        """Test parameterized config without input"""
+        (tmp_path / "local").mkdir(exist_ok=True)
+
+        catalog = TemplatedConfigLoader(conf_paths).get("catalog*.yml")
+
+        assert catalog["boats"]["type"] == "${boat_data_type}"
+        assert (
+            catalog["boats"]["filepath"]
+            == "${s3_bucket}/${raw_data_folder}/${boat_file_name}"
+        )
+        assert catalog["boats"]["columns"]["id"] == "${string_type}"
+        assert catalog["boats"]["columns"]["name"] == "${string_type}"
+        assert catalog["boats"]["columns"]["top_speed"] == "${float_type}"
+        assert catalog["boats"]["users"] == ["fred", "${write_only_user}"]
+
+    @pytest.mark.usefixtures("proj_catalog_advanced")
+    def test_catlog_advanced(self, tmp_path, conf_paths):
+        """Test whether it responds well to advanced yaml values (i.e. nested dicts, booleans,
+        lists, etc.)"""
+        (tmp_path / "local").mkdir(exist_ok=True)
+
+        catalog = TemplatedConfigLoader(conf_paths).get("catalog*.yml")
```

Comment:
I'd suggest to pass globals dict even if there is nothing to replace since it would break the current flow as it can't handle non-string values (see my comment above)